pkg/controller/datavolume-controller.go

```diff
@@ -2722,3 +2815,294 @@ func getDeltaTTL(dv *cdiv1.DataVolume, ttl int32) time.Duration {
 	}
 	return delta
 }
+
+// detectCloneSize obtains and assigns the original PVC's size when cloning using an empty storage value
+func (r *DatavolumeReconciler) detectCloneSize(
+	dv *cdiv1.DataVolume,
+	pvcSpec *corev1.PersistentVolumeClaimSpec,
+	cloneType cloneStrategy) (bool, error) {
+
+	var targetSize resource.Quantity
+	sourcePvc, err := r.findSourcePvc(dv)
+	if err != nil {
+		return false, err
+	}
+	sourceSize := sourcePvc.Status.Capacity.Storage()
+
+	// Due to possible filesystem overhead complications when cloning
+	// using host-assisted strategy, we create a pod that automatically
+	// collects the size of the original virtual image with 'qemu-img'.
+	// If another strategy is used or the original PVC's volume mode
+	// is "block", we simply extract the value from the original PVC's spec.
+	if cloneType == HostAssistedClone &&
+		getVolumeMode(sourcePvc) == corev1.PersistentVolumeFilesystem &&
+		GetContentType(sourcePvc) == string(cdiv1.DataVolumeKubeVirt) {
+		// If available, we first try to get the virtual size from previous iterations
+		imgSize, available := getSizeFromAnnotations(sourcePvc)
+		if !available {
+			imgSize, err = r.getSizeFromPod(sourcePvc, dv)
+			if err != nil {
+				return false, err
+			} else if imgSize == 0 {
+				return false, nil
+			}
+		}
+		// Allow the clone-controller to skip the size comparison requirement
+		// if the source's size ends up being larger due to overhead differences
+		if sourceSize.CmpInt64(imgSize) == 1 {
+			dv.Annotations[AnnPermissiveClone] = "true"
+		}
+		// If needed, inflate size with filesystem overhead
+		targetSize, err = inflateSizeWithOverhead(r.client, imgSize, pvcSpec)
+		if err != nil {
+			return false, err
+		}
+	} else {
+		targetSize = *sourceSize
```

Comment:
here we should also inflate with correct overhead if the target is filesystem!

Reply:
You are right, I'll add a new commit with the requested changes. Thanks!